### DISCUSSION TITLE
Ultimaker 3 compatibility

### DISCUSSION THREAD

**Discussion by sstepane** [FormerLurker/ArcWelderPlugin#231] [Q&A]
Hello,

Is Arc Welder works well with Ultimaker 3? If so - what should be default settings used?
Thanks!

**Comment by FormerLurker** [comment]
The default settings should be good depending on how the firmware handles arc commands.  I would recommend trying to print something small, like a benchy, to verify that ultimaker firmware can handle arcs.  Im not sure if their firmware is open or closed source, but if you can point me to the source code, I would be happy to take a look!

**Comment by seantapscott** [comment]
It doesn't handle arc commands well last time I tried on UM3.  Limited
value as well as the compression of the gcode files isn't really necessary.


I searched around and found this, which appears to be the official source
code of the UM3 marlin branch.  I think It's very custom, but I can't
actually tell.
https://github.com/Ultimaker/UltimakerMarlin/tree/Ultimaker3

**Reply by FormerLurker** [reply]
Thanks for the link!  It is using the old marlin 1 algorithm that only uses mm_per_arc_segment.  It is set to 1mm, which will produce flat edges for very small arcs.  With firmware compensation enabled, it should work ok, but not as good as firmware with better arc support.

**Comment by seantapscott** [comment]
Yeah, I feel like that might not be what they're actually using, as the
number of extruders is set to 1, which is very false, as UM3 is, by
default, a dual extruder machine.  Anyway, it still might relevant to arcs,
as their marlin implementation is old.
